### ISSUE TITLE
Template espèces : ajouter recherches Wikipedia, Google, ...

### ISSUE THREAD

**Issue by sig-pnrnm** [PnX-SI/GeoNature-atlas#188]
Il serait pratique, sur les fiches espèces, de disposer de liens de recherche sur Wikipedia, Google (ou autres).

Je viens de le faire rapidement sur notre instance avec ce code :
```html
<a class="btn btn-default" height="30" style="border:1px solid silver; text-align:left;" 
href="https://fr.wikipedia.org/w/index.php?search={{taxon.taxonSearch.lb_nom}}"
role="button" target="_blank" data-toggle="tooltip"
data-original-title="Recherche Wikipedia" data-placement="bottom">
<i class="fa fa-wikipedia-w" aria-hidden="true"></i>&nbsp;Wikipedia
</a>
<a class="btn btn-default" height="30" style="border:1px solid silver; text-align:left;" 
href="https://www.google.fr/search?q={{taxon.taxonSearch.lb_nom}}"
role="button" target="_blank" data-toggle="tooltip"
data-original-title="Recherche Google" data-placement="bottom">
<i class="fa fa-google" aria-hidden="true"></i>&nbsp;Google
</a>
```
Ce qui donne quelque chose comme ça :
![recherches](https://user-images.githubusercontent.com/26482054/57293401-eaff1f80-70c4-11e9-9974-78ce2e9f305e.png)

Après, dans quelle section placer ces boutons ?

**Comment by sig-pnrnm** [comment]
Et si on veut que ce soit plus petit, juste les icones suffisent :
![recherches2](https://user-images.githubusercontent.com/26482054/57294479-1c2d1f00-70c8-11e9-9eb0-33305faca076.png)

**Comment by camillemonchicourt** [comment]
On a évoqué le fait de pouvoir récupérer et intégrer du contenu tiers à chaud, par exemple une description Wikipedia. 
Par contre faire des boutons pour renvoyer vers des recherches Google, je vois pas bien l'intérêt. 
Si les gens veulent aller sur Google, ils savent faire. 
Sachant en plus, que ça fait partir les visiteurs et que c'est pénalisant en terme de référencement.

**Comment by sig-pnrnm** [comment]
>  Si les gens veulent aller sur Google, ils savent faire.

Bien sûr, je ne suis pas le public cible : je réponds en tant que naturaliste.
Je pratique beaucoup l'Atlas dans me recherches naturalistes perso, et je passe pour cela d'un site à l'autre. Et ces boutons, depuis que je les ai mis en place, me rendent beaucoup service.
Tant qu'ils sont discrets (dans un onglet masqué au chargement), je ne trouve pas qu'ils perturbent la page.

Pour ce qui est des espèces entomo, il y a d'ailleurs un site de référence que je vais ajouter, c'est la galerie du "Monde des insectes". Ils ont un moteur de recherche taxonomique sous cette forme : 
https://www.galerie-insecte.org/galerie/wikige.php?tax=taxon ([exemple](https://www.galerie-insecte.org/galerie/wikige.php?tax=Zygaena%20fausta)).
(par contre, je sais pas comment ajouter une condition pour que ce bouton ne s'affiche que pour les fiches "invertébrés")

Je me doute qu'on va revenir à l'éternel débat : "le naturaliste n'est pas le public cible". Mais faute d'outil à disposition du Naturaliste, les Atlas sont incontournablement utilisés par ces gens peu fréquentables (dont je suis :wink: )

> c'est pénalisant en terme de référencement.

Ca s'ouvre dans un autre onglet, donc les visiteurs ne quittent pas l'Atlas.

**Comment by sig-pnrnm** [comment]
> par contre, je sais pas comment ajouter une condition pour que ce bouton ne s'affiche que pour les fiches "invertébrés"

Je m'auto répond, c'était tout bête en fait, avec une condition `{% if taxon.taxonSearch.group2_inpn == 'Insectes' %}` : 
```html
{% if taxon.taxonSearch.group2_inpn == 'Insectes' %}
	<a class="btn btn-default" height="30px" width="60px"
	style="border:1px solid silver; text-align:left;"
	href="https://www.galerie-insecte.org/galerie/wikige.php?tax={{taxon.taxonSearch.lb_nom}}"
	role="button" target="_blank" data-toggle="tooltip"
	data-original-title="Galerie du Monde des Insectes"
	data-placement="bottom">
	<i class="fa fa-bug" aria-hidden="true"></i>
	</a>
{% else %}
{% endif %}
```
Libre à chacun de l'ajouter à son template, selon son usage / public cible ;)

**Comment by sig-pnrnm** [comment]
Question pour ceux qui manipulent Symphony et/ou PHP.
N'y aurait-il pas plus propre que cette syntaxe :
```php
{%	if	taxon.taxonSearch.group2_inpn == 'Insectes'
	or	taxon.taxonSearch.group2_inpn == 'Arachnides'
	or	taxon.taxonSearch.group2_inpn == 'Crustacés'
	or	taxon.taxonSearch.group2_inpn == 'Entognathes'
	or	taxon.taxonSearch.group2_inpn == 'Myriapodes'
%}
```
En SQL, je me sers de `champ IN ('valeur 1', 'valeur 2', ...)`, mais en PHP je ne vois pas comment raccourcir cette condition.

**Comment by camillemonchicourt** [comment]
GN-atlas est développé en Python (framework Flask).

**Comment by sig-pnrnm** [comment]
Merci de m'avoir ré-orienté dans mes recherches, mais même avec le bon language, je ne vois pas comment écrire cette condition plus "proprement" (mais peut-être est-ce déjà suffisament propre : ça marche en tout cas !)

**Comment by TheoLechemia** [comment]
Pour être encore plus précis, le moteur de template utilisé par flask est Jinja. Tu peux regarder du côté de la doc, mais pas sûr qu'il y ai une synthaxe plus épurée.

**Comment by sig-pnrnm** [comment]
> pas sûr qu'il y ai une synthaxe plus épurée.

Si même Théo ne peut faire plus épuré, alors je vais considérer que la syntaxe est correcte, puisqu'elle fonctionne ! :wink:

**Comment by xavyeah39** [comment]
Salut,
Si je ne m'abuse cette syntaxe fonctionne aussi : 
```python
{%  if taxon.taxonSearch.group2_inpn in ['Insectes','Arachnides','Crustacés','Entognathes','Myriapodes'] %}
```

**Comment by laurentbarthe** [comment]
> Bien sûr, je ne suis pas le public cible : je réponds en tant que naturaliste.
> Je pratique beaucoup l'Atlas dans me recherches naturalistes perso, et je passe pour cela d'un site à l'autre. Et ces boutons, depuis que je les ai mis en place, me rendent beaucoup service.

Je te rejoins à 100% @sig-pnrnm. Nous ne sommes peut être pas l'unique cible mais en majorité ce sont quand même les naturalistes qui utilisent ces outils. Si on peut leur faciliter un peu le taf c'est cool car c'est un peu la jungle pour trouver de l'information de qualité.

**Comment by DonovanMaillard** [comment]
Pour le coup, je suis plutôt d'accord avec ce que dit Camille, ces boutons ne servet pas forcement à mon avis, même pour un naturaliste... 

Soit je veux des infos sur l'espèce, et dans ce cas je cherche déjà sur google (qui me renvoie sur wikipédia ou éventuellement sur l'atlas). Soit je veux des infos dans mon territoire, et dans ce cas je vais sur l'atlas... 

Dans tous les cas si ca apporte une réponse aux besoins de certains, il n'y a pas de raison de ne pas le faire, mais c'est le genre de choses que je mettrais en paramètre pour laisser le choix à chacun de l'activer ou non sur son instance :)

**Comment by TheoLechemia** [comment]
Regardez, le bouton google existe déjà !!
![google](https://user-images.githubusercontent.com/15265745/61887308-ad5ea580-af01-11e9-8630-4f70c21abb04.png)

Pas non plus hyper fan des de l'alourdissement des interfaces...
Vivement le comité utilisateur GeoNature qu'on puisse discuter de tout ça !
